pkg/segment/segment.go

```diff
@@ -0,0 +1,132 @@
+package segment
```

Comment:
I feel there needs to be an interface TelemetryClient with Segment as an implementation.
The methods are
InitClient
Upload

getUserIdentity is common for all clients.

Reply:
@prashbnair, in the Telemetry package, we are maintaining a context state. Now if we convert it to the interface, the whole point of having it in the first place is kinda repurposed. We are coding inline with crc patterns